azext_iot/_validators.py

```diff
@@ -36,3 +37,8 @@ def mode2_iot_login_handler(cmd, namespace):
 
             if not any([login_value, entity_value, offline]):
                 raise RequiredArgumentMissingError(error_no_hub_or_login_on_input(iot_cmd_type))
+
+
+def process_top(namespace):
+    if hasattr(namespace, "top"):
+        namespace.top = _process_top(top=namespace.top)
```

Comment:
The integer validation for the top argument happens before this event/validator handler right? As in there is no chance of a non-integer being evaluated here?

Reply:
![image](https://user-images.githubusercontent.com/73560279/218824929-5835d3d0-05a3-4f45-aa66-9fa95bbc0cab.png)
yes